src/lib.rs

```diff
@@ -72,19 +70,121 @@ struct PointProps<'a> {
     pub n_spatial_dims: usize,
 }
 
+impl PointProps<'_> {
+    pub fn get_weight(&self, idx: usize) -> f64 {
+        if let Some(weights) = self.weights {
+            weights[idx]
+        } else {
+            1.0
+        }
+    }
+}
+
+// TODO use binary search, and have specialized version for regularly spaced bins?
```

Comment:
I was taking a really quick glance and I noticed this comment.

I feel like I left the exact same comment when I had implemented this function in C++. Just curious, were you looking at my code when you wrote this or is it a complete coincidence?

Reply:
Coincidence! It's also a similar problem to some of the wavelength windowing stuff that Korg does, which was working on lately.

Reply:
Honestly, the more I think about it, the less clear it is that binary search helpful.  We'll have to run some benchmarks later.